### PR TITLE
udev: include USB revision in ID_PATH

### LINKED ISSUE #19406: udev assigns the same ID_PATH for different usb devices (Built-in USB camera and External USB 3.0 camera via USB-C port)
**systemd version the issue has been seen with**
 > 247

**Used distribution**
 > …

**Linux kernel version used** (`uname -a`)

 > `Linux localhost 5.4.113-14419-g47d1a7be3a65 #1 SMP PREEMPT Thu Apr 22 12:49:28 CST 2021 x86_64 Intel(R) Core(TM) i5-7Y57 CPU @ 1.20GHz GenuineIntel GNU/Linux`

**CPU architecture issue was seen on**
 > x86_64

**Expected behaviour you didn't see**
 > …

**Unexpected behaviour you saw**
 > The same ID_PATH are given for built-in USB camera and the external USB 3.0 camera (Connected via USB-C port) while they are both connected.

**Steps to reproduce the problem**
 > I observed the issue on a Google Pixelbook using the USB-C port on the right side.

Built-in Camera:
```
$ udevadm info --query=all /dev/video0
...
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb1/1-2/1-2:1.0/video4linux/video0
...
E: ID_PATH=pci-0000:00:14.0-usb-0:2:1.0
```

External Camera:
```
$ udevadm info --query=all /dev/video2
...
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb2/2-2/2-2:1.0/video4linux/video2
...
E: ID_PATH=pci-0000:00:14.0-usb-0:2:1.0
```

There is a similar issue which is filed a few years ago:
https://github.com/systemd/systemd/issues/10683

Here are some extra outputs which are asked to provide in that issue:
[Output of $ lspci -nnv](https://gist.github.com/williamiced/4a3f3f3b9a3a79a5f49d36313b3aacc3)
[Output of $ lsusb -v](https://gist.github.com/williamiced/097334b4f61e5ceb5713dbde99225b8f)
[Output of $ udevadm info --export-db](https://gist.github.com/williamiced/1d2a0d4e55990a1e972b2875475ecf8e)
[Output of $ ls -lR /sys](https://gist.github.com/williamiced/7b9e7c1b081bdf894fc20a461f4996cd)

Please let me know if you need any more information.
Thanks for helping.

### PR DESCRIPTION
**Pull request by yuwata**
Closes #19406.

### CODE REVIEW

File: rules.d/60-drm.rules [hunk 1]
```diff
@@ -3,6 +3,9 @@
 ACTION!="remove", SUBSYSTEM=="drm", SUBSYSTEMS=="pci|usb|platform", IMPORT{builtin}="path_id"
 
 # by-path
-ENV{ID_PATH}=="?*", KERNEL=="card*", SYMLINK+="dri/by-path/$env{ID_PATH}-card"
-ENV{ID_PATH}=="?*", KERNEL=="controlD*", SYMLINK+="dri/by-path/$env{ID_PATH}-control"
-ENV{ID_PATH}=="?*", KERNEL=="renderD*", SYMLINK+="dri/by-path/$env{ID_PATH}-render"
+KERNEL=="card*",     ENV{ID_PATH}=="?*",            SYMLINK+="dri/by-path/$env{ID_PATH}-card"
+KERNEL=="card*",     ENV{ID_PATH_USB_COMPAT}=="?*", SYMLINK+="dri/by-path/$env{ID_PATH_USB_COMPAT}-card"
```
Comment (bluca) [inline rules.d/60-drm.rules]: mmh will this break existing software? If so, shouldn't it be the other way around, and a new ENV added that includes the protocol revision, leaving the existing unchanged?

Comment (keszybz) [reply rules.d/60-drm.rules]: We discussed this during a call today, and I also think that adding a new variable would be safer.

http://codesearch.debian.net/search?q=ID_PATH&literal=1&perpkg=1 shows that there are some packages which look at ID_PATH. Also, people can have custom unpackaged rules.

**Comment by dtardon** [comment]
> So, there are three options:
> 
>     * introduce ID_PATH_WITH_USB_REVISION or so which include USB revision, or
> 
>     * make versioning for network interface naming to more generic and add a new flag for ID_PATH,
> 
>     * introduce versioning for ID_PATH.
> 
> 
> Which one do you like?
> 
> @dtardon I thought you commented about introducing versioning in the last meeting, but I could not follow at that time. Could you rephrase that?

No, that was @lnykryn. He suggested introducing naming schemes for ID_PATH, but separately from network naming schemes, IIUC.

**Review by keszybz — APPROVED** [review]
LGTM.

This is all a bit verbose, but this approach maximizes backwards compatibility. It's probably safest to do resolve this issue with a new property and new paths, keeping the old stuff in place.

**Comment by keszybz** [comment]
jammy-ppc64el:
```
TEST-29-PORTABLE:                   FAIL     (1809 s)
TEST-67-INTEGRITY:                  FAIL     (175 s)
TEST-70-TPM2:                       FAIL     (154 s)
```
Looks like machine overload, unrelated to the those patches.

**Comment by poettering** [comment]
lgtm btw
